### ISSUE TITLE
Publish artifacts (tar.gz/zips) to maven or an S3 bucket living within the repo managed account

### ISSUE THREAD

**Issue by dblock** [opensearch-project/opensearch-build#114]
**Is your feature request related to a problem? Please describe.**

As part of #20, publish artifacts (tar.gz/zips) to an S3 bucket living within the repo managed account.

- [ ] Document how to create and secure an S3 bucket, e.g. naming convention, folder structure, write for the repo account, read for bundle building account.
- [ ] Publish artifacts from GH workflow, include a manifest that refers back to commit SHA in the artifact
- [ ] Provide a reference implementation

**Comment by mch2** [comment]
I think we should reword this to:
Update to include plugin zip as a maven publication.  When this is done these zips will get picked up and published with #20 without the need for extra S3 buckets.

**Comment by dblock** [comment]
> I think we should reword this to:
> Update to include plugin zip as a maven publication. When this is done these zips will get picked up and published with #20 without the need for extra S3 buckets.

I renamed this to say "to maven or S3" this way if the maven idea doesn't work we still know we have the S3 option

**Comment by frotsch** [comment]
I would strongly recommend to publish to maven instead of S3 because maven guarantees immutability of the artifacts and is more "standard" and "convenient" for artifact consumption.

**Comment by mch2** [comment]
Closing this as a duplicate of issue #20
